src/ngSanitize/filter/linky.js

```diff
@@ -137,8 +163,21 @@ angular.module('ngSanitize').filter('linky', ['$sanitize', function($sanitize) {
     }
 
     function addLink(url, text) {
+      var key;
+      var computedAttributes = {};
       html.push('<a ');
-      if (angular.isDefined(target)) {
+      if (angular.isObject(attributes)) {
```

Comment:
Can this code duplication be prevented using something like this?

```js
var key;
var computedAttributes = {};
if (angular.isFunction(attributes)) {
  attributes = attributes(url);
}
if (angular.isObject(attributes)) {
  computedAttributes = attributes;
  for (key in computedAttributes) {
    html.push(key + '="' + attributes[key] + '" ');
  }
}
```

This would also check if the return value from the `attributes` function is an object.